src/checksum.h

```diff
@@ -130,6 +130,7 @@ static inline rs_weak_sum_t weaksum_digest(weaksum_t *sum)
     case RS_RABINKARP:
         return rabinkarp_digest(&sum->rk);
     }
+    return 0;
```

Comment:
I wonder if these fragments would be faster in the following form;
```C
  if (sum->kind == RS_ROLLSUM)) 
    return mix32(RollsumDigest(&sum->rs));
  return rabinkarp_digest(&sum->rk);
```
The main reason why I didn't code them all that way is it doesn't extend as nicely if we ever add another rollsum type. However, I suspect we will never add another rollsum... RabinKarp is pretty close to optimal on collisions and clustering.

Reply:
Or use `default:` if you want to keep `switch` in code...